### ISSUE TITLE
Bad Owner or Permissions on username\.ssh\config file

### ISSUE THREAD

**Issue by mashirozx** [microsoft/vscode-docs#3210]
When using vscode-remote it shows `C:\\Users\\username/.ssh/config`.

Tried to fix following the doc at <https://github.com/microsoft/vscode-docs/blob/master/docs/remote/troubleshooting.md#fixing-ssh-file-permission-errors>

But the command seems not work for me:

![image](https://user-images.githubusercontent.com/16148054/68179818-54a4ab80-ffcc-11e9-9daa-e6775796036f.png)

**Comment by mashirozx** [comment]
@Chuxel

**Comment by Chuxel** [comment]
@mashirozx The issue you reference was in fact resolved with this script, so if the script isn't running, that's something else.  

Are you using the Windows OpenSSH client in Windows 10 1803+? If not, you can try to manually set the permissions on `config` and other files in the `.ssh` folder to SYSTEM, Administrators, and your user instead.

**Comment by mashirozx** [comment]
@Chuxel 
System: Windows 10 Home 1903
OS build: 18362.418

Actually I solved this problem in vscode-remote by modifying `remote.SSH.configFile` and `remote.SSH.path` and copying `C:\Windows\System32\OpenSSH\` to my own directory. I think this could be showed in the docs (maybe not recommend but it works and is really easy) since the PowerShell scripts seems not really work.

**Comment by Chuxel** [comment]
@mashirozx I'd rather not include something without an understanding of what the problem was to begin with - the configFile is covered elsewhere. 

This seems to be a signature error which shouldn't be possible. The only thing I can think of is you are on Home instead of Pro or Enterprise. This is the first time this error has been reported, so something strange is going on.

**Comment by mashirozx** [comment]
@Chuxel Maybe this should be a new issue in https://github.com/PowerShell/Win32-OpenSSH/? 😂

**Comment by mashirozx** [comment]
@Chuxel Actually there is also an issue relate to the Windows version (I guess) https://github.com/PowerShell/Win32-OpenSSH/issues/1445

**Comment by Chuxel** [comment]
Yeah that looks like the issue! Let's close this as a upstream duplicate of that problem.  Since its not related to VS Code.

**Comment by hacksut3l** [comment]
@Chuxel I also encountered this problem. Is it because of the path?
![image](https://user-images.githubusercontent.com/11365367/70321494-6a72ee00-1862-11ea-95fa-4a19cd8a6b3d.png)
Instead of "C:\\Users\\username\\.ssh\\config"

**Comment by mmmeff** [comment]
@hacksut3l I fixed this by setting the path in my ssh config file to `C:/Users/[name]/.ssh/config` or wherever your config file lives. The important part here is to use forward slashes, not the windows standard backslashes. I think this is because the Windows 10 OpenSSH client is resolving paths with PowerShell under the hood which supports both path delimiters.

**Comment by mashirozx** [comment]
@mmmeff That is the problem!:+1:

**Comment by prashanthc97** [comment]
@mmmeff Spent 2 hours finding the issue, then came across this. Worked like a charm!

**Comment by Atem18** [comment]
Can confirm guys, the solution of @mmmeff is working. Should not we open another issue to fix it since by default the extension does not work.

**Comment by karpulix** [comment]
![Screenshot_34](https://user-images.githubusercontent.com/16068884/81641790-0923fc00-944c-11ea-9cb9-e13d9a2facc5.png)
I resolved this problem when added obvious path for .ssh config as "C:\\User\\Alexander\\.ssh\\config" (use double slash). It is working for me.

**Comment by phuonghx** [comment]
> ![Screenshot_34](https://user-images.githubusercontent.com/16068884/81641790-0923fc00-944c-11ea-9cb9-e13d9a2facc5.png)
> I resolved this problem when added obvious path for .ssh config as "C:\User\Alexander\.ssh\config" (use double slash). It is working for me.

Thanks!

**Comment by iswangaiguo** [comment]
> ![Screenshot_34](https://user-images.githubusercontent.com/16068884/81641790-0923fc00-944c-11ea-9cb9-e13d9a2facc5.png)
> I resolved this problem when added obvious path for .ssh config as "C:\User\Alexander\.ssh\config" (use double slash). It is working for me.

thank you very much，you solved my problems

**Comment by vinvinodk** [comment]
Those who are using recent Windows 10 (18.03+ Update and above) will be having OpenSSH installed. You need to uninstall that feature by going to "manage optional features" in Apps and Features menu from Settings. Once uninstalled, install/keep the Git for Windows. It will solve the problem.

> Windows 10 1803+ - UnInstall the Windows OpenSSH Client and Install [Git for Windows](https://git-scm.com/download/win)
> Earlier Windows -  Install [Git for Windows](https://git-scm.com/download/win)

No file permissions issue hereafter. enjoy. 

P.S. I ran into this issue while doing the Windows Update to 1803 and solved by uninstalling openSSH

**Comment by conix67** [comment]
This fixed the same issue for me - https://github.com/Microsoft/vscode-remote-release/issues/119

**Comment by Stevemaster92** [comment]
> ![Screenshot_34](https://user-images.githubusercontent.com/16068884/81641790-0923fc00-944c-11ea-9cb9-e13d9a2facc5.png)
> I resolved this problem when added obvious path for .ssh config as "C:\User\Alexander\.ssh\config" (use double slash). It is working for me.

Thanks my friend! 👍  This even works without uninstalling the OpenSSH Client on Windows

**Comment by zxbuaa** [comment]
I suddenly encountered this issue after login Windows with another account and then login back. Before that I didn't touch the default path of .ssh/config
Is this <code>BUG</code> of vscode or specific version of Windows 10?

**Comment by doxxx** [comment]
I believe this issue should be re-opened since the _default_ configuration is broken. This should just work out of the box.

**Comment by m3hdad** [comment]
> 
> 
> @hacksut3l I fixed this by setting the path in my ssh config file to `C:/Users/[name]/.ssh/config` or wherever your config file lives. The important part here is to use forward slashes, not the windows standard backslashes. I think this is because the Windows 10 OpenSSH client is resolving paths with PowerShell under the hood which supports both path delimiters.
> 
> For example:
> 
> ```
> Host ExampleHost
>   HostName 1.2.3.4
>   IdentityFile C:/Users/johndoe/.ssh/id_rsa
>   User johndoe
> ```

On my system the issue was resolved by:

1. setting the IdentityFile path as reported above.
2. defining custom config path for applications like vs code

**Comment by MrMarshall** [comment]
For me this was caused by the default ssh client. I got the same error when I used ssh through cmd.exe.

What solved it was using a different ssh client, e.g. the git bash ssh client. Add the following entry to your settings.json file:
`"remote.SSH.path": "C:/Program Files/Git/usr/bin/ssh.exe"`

**Comment by solinad21** [comment]
@MrMarshall Thanks, now run OK!

**Comment by timeneverdie** [comment]
I just fix this problem. Let me tell you the real reason: **Don't use Putty Key Generator!**

After OpenSSH was built in Windows 10, VSCode will always use the standard key format.

Open your terminal, use the command `ssh-keygen -t rsa -b 2048 -C "username"` (must be a Capitalized character C)

Then use your file manager or cd in Terminal to find out the default SSH folder on Windows 10 `C:\Users\Username\.ssh`. You will find the private key generated just now has `no extension`. But if you use Putty Key Generator, your private key will have a `*.ppk` extension. That's the real cause for Key Denied!

VSCode will use the default folder `C:\Users\Username\.ssh` for you, so you don't need to write any SSH Path into the config file. But if you use another SSH library, maybe you need to edit something. (When you follow the GUI of VSCode, you will see the default SSH folder. Be careful with the `\` mark.)

The last thing to do is upload the `*pub` file to the server (or open with VSCode to copy and paste into VPS Console). When done, you can login to the server directly. The information included in the `config` file will be used automatically. 

Make sure your `config` file has these fields:

```
Host       easy_edit_to_show_in_the_tree_panel    #no need to use quote mark
HostName   ipaddress_of_the server                #number                     
User       generated_with_-C                      #no need to use quote mark
Port       port_you_set, default_is_22            #number
```

**Comment by doxxx** [comment]
I think I found an alternative solution to this issue in this Stackoverflow answer: https://stackoverflow.com/a/58275268/345722

**Comment by skeddles** [comment]
This took me forever to figure out. 

First I had to use the suggestion from here: https://github.com/microsoft/vscode-docs/issues/3210#issuecomment-569464023
Which was going to the settings of the extension and setting the custom config file path to the `./ssh/config` file (making sure there are forward slashes only).

I then also had to change the permissions on the key file I was using, described here: https://superuser.com/a/1296046/302150

Ultimately this was way too hard to figure out, I've wasted hours. If you for some reason can't fix the fact that the default path doesn't work at all, you could at least provide error messages that actually explain the problem and tell us how to fix it.

**Comment by tmontney** [comment]
> ![Screenshot_34](https://user-images.githubusercontent.com/16068884/81641790-0923fc00-944c-11ea-9cb9-e13d9a2facc5.png) I resolved this problem when added obvious path for .ssh config as "C:\User\Alexander\.ssh\config" (use double slash). It is working for me.

Still an issue, and this fixed it.

VSC: 1.61.2
Remote-SSH: 0.65.8

**Comment by mldisibio** [comment]
> I think I found an alternative solution to this issue in this Stackoverflow answer: https://stackoverflow.com/a/58275268/345722

Happened to me right after IT Support logged in as Admin. Removing them (and disabling inheritance) using Security dialog on ssh folder fixed the issue.

**Comment by choncba** [comment]
Solved for me adding absolute path to ssh config file in vs code ssh settings.

**Comment by inamahamed** [comment]
All Idiots here. none of them could give an answer to this issue

**Comment by kangarooo** [comment]
Needed to edit myself to get rid of it
![image](https://user-images.githubusercontent.com/73187/160458075-186e2a1d-d295-45d6-b6a1-d46216b7d61b.png)

**Comment by xDaniCh** [comment]
> Пришлось отредактировать себя, чтобы избавиться от него ![изображение](https://user-images.githubusercontent.com/73187/160458324-e47b93e3-c489-41d4-b93b-de162382ceb5.png)

Great, that really helped.

**Comment by xiay-lcw** [comment]
In my case, the file permission was indeed corrupted.  I tried to manually fix the permission on the file but did not work.  I guess there are hidden permission properties `openssh` would check for but not exposed in windows explorer GUI.

I ended up creating a new file and copy over the content...

**Comment by Electromorphous** [comment]
I'm on ubuntu and my terminal was working perfectly fine with git and the ssh config file. I was able to clone as well as push code. But the vs code integrated terminal kept giving the "Bad owner or permissions" error.

None of the above-stated solutions was helpful to me as I'm on Ubuntu. Therefore I found [this](https://superuser.com/questions/1212402/bad-owner-or-permissions-on-ssh-config-file).

Basically, I just ran the 2 commands given below
```
chmod 700 ~/.ssh
chmod 600 ~/.ssh/*
```

**Comment by advra** [comment]
> Needed to edit myself to get rid of it ![image](https://user-images.githubusercontent.com/73187/160458324-e47b93e3-c489-41d4-b93b-de162382ceb5.png)

This is a great alternate solution if one does not have admin privs.

**Comment by pimentelleo** [comment]
Got this error while trying to connect into a server that uses a pubkey.
I solved on my side by changed the permissions of the file itself to only me full access and it worked.

**Comment by Vaarun-Kamath** [comment]
> ![Screenshot_34](https://user-images.githubusercontent.com/16068884/81641790-0923fc00-944c-11ea-9cb9-e13d9a2facc5.png) I resolved this problem when added obvious path for .ssh config as "C:\User\Alexander\.ssh\config" (use double slash). It is working for me.

OMG IT WORKS 😭

**Comment by CristiFati** [comment]
@mmef solution (https://github.com/microsoft/vscode-docs/issues/3210#issuecomment-569464023) works for me with the following notes:
 
- It didn't work for a key file located somewhere on my *E:* drive
 
- I had to *SymLink* it in my home *dir* (`mklink C:\Users\cfati\.ssh\key_file_lnk E:\${path_to_actual_key_file}`)
